test/unit/pipeline/FetchUnitTest.cc

```diff
@@ -437,5 +453,238 @@ TEST_F(PipelineFetchUnitTest, idleLoopBufferDueToNotTakenBoundary) {
   EXPECT_EQ(output.getTailSlots()[0], mOp2);
 }
 
+// Tests that a compressed instruction held at the end of the fetch buffer is
```

Comment:
The term compressed doesn't apply very well here. The fetch unit is agnostic to the ISA and thus has no concept of a compressed instruction. We should refer to such instances in relation to the "minimum instruction size" rather than "compressed"